### PR TITLE
Better resource management for Artifactory

### PR DESCRIPTION
**Pull request by eldada**
* Update Artifactory version
* Add resources requests and limits to all pods
* Ability to set whitelist on Nginx LoadBalancer service
* Support setting a custom artifactory service name (that will also be set in nginx config)
* Add new parameters to README
* Fix artifactory replicas reference. Hardcode database replicas to 1
* Use resource memory requests and limits in Artifactory java options
* Add readinessProbe to Artifactory
* Add readme on creating a Kubernetes Docker registry secret and passing it to helm

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

File: stable/artifactory/values.yaml [hunk 1]
```diff
@@ -37,6 +39,13 @@ database:
     ##   GKE, AWS & OpenStack)
     ##
     # storageClass: "-"
+  resources:
```
Comment (unguiculus) [inline stable/artifactory/values.yaml]: We recommend not to specify defaults for resources. This should be a conscious choice of the user. Plus, it makes it easier to test charts on small systems, such as Minikube. Do it like so:

```yaml
resources: {}
  # requests:
    # memory: "512Mi"
    # cpu: "100m"
  # limits:
    # memory: "1Gi"
    # cpu: "500m"
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by eldada** [comment]
@unguiculus - I added the requested changes.

File: stable/artifactory/README.md [hunk 2]
```diff
@@ -103,6 +155,10 @@ The following tables lists the configurable parameters of the artifactory chart
 | `nginx.persistence.enabled` | Nginx persistence volume enabled | `true`   |
 | `nginx.persistence.accessMode` | Nginx persistence volume access mode | `ReadWriteOnce`   |
 | `nginx.persistence.size` | Nginx persistence volume size | `5Gi`   |
+| `nginx.resources.requests.memory` | Nginx initial memory request  | `250Mi`   |
+| `nginx.resources.requests.cpu`    | Nginx initial cpu request     | `100m`    |
+| `nginx.resources.limits.memory`   | Nginx memory limit            | `250Mi`   |
+| `nginx.resources.limits.cpu`      | Nginx cpu limit               | `500m`    |
```
Comment (unguiculus) [inline stable/artifactory/README.md]: Defaults are not correct.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by eldada** [comment]
@unguiculus - done! Default values corrected.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**